Sources/TuistCore/Cache/CacheOutputType.swift

```diff
@@ -8,36 +14,79 @@ public enum CacheOutputType: CustomStringConvertible {
     /// Frameworks built for the simulator.
     case framework
 
-    /// XCFrameworks built for the simulator and device.
-    case xcframework
-
-    /// XCFrameworks built for devices.
-    case deviceXCFramework
-
-    /// XCFrameworks built for simulators.
-    case simulatorXCFramework
+    /// XCFrameworks built for the simulator and/or device.
+    case xcframework(CacheXCFrameworkType?)
 
     public var description: String {
         switch self {
         case .bundle:
             return "bundle"
         case .framework:
             return "framework"
-        case .xcframework:
-            return "xcframework"
-        case .deviceXCFramework:
-            return "device-xcframework"
-        case .simulatorXCFramework:
-            return "simulator-xcframework"
+        case let .xcframework(type):
+            switch type {
+            case .device:
+                return "device-xcframework"
+            case .simulator:
+                return "simulator-xcframework"
+            case nil:
+                return "xcframework"
+            }
         }
     }
+}
 
+extension CacheOutputType {
     public var isXCFramework: Bool {
         switch self {
         case .bundle, .framework:
             return false
-        case .xcframework, .deviceXCFramework, .simulatorXCFramework:
+        case .xcframework:
+            return true
+        }
+    }
+
+    public var shouldBuildForSimulator: Bool {
+        switch self {
+        case .bundle:
+            return false
+        case .framework:
             return true
+        case let .xcframework(type):
+            return type == nil || type == .simulator
+        }
+    }
+
+    public var shouldBuildForDevice: Bool {
+        switch self {
+        case .bundle:
+            return false
+        case .framework:
+            return true
+        case let .xcframework(type):
+            return type == nil || type == .device
+        }
+    }
+}
+
+extension CacheOutputType: Equatable {
+    public static func == (lhs: Self, rhs: Self) -> Bool {
+        switch lhs {
```

Comment:
I would do `switch (lhs, rhs)`, also, isn't this automatically synthesized by the compiler 🤔

Reply:
Oh yes, I forgot that Swift now auto synthesizes equatable for enum with associated value 😁